recipe/meta.yaml

```diff
@@ -1,75 +1,139 @@
-    - typing_extensions >=3.10
-    - tomli >=1.1.0
-    # extra: dmypy
-    - psutil >=4.0
 
-test:
-  imports:
-    - mypy
-    - mypy.plugins
-    - mypy.server
-    - mypy.dmypy
-  commands:
-    - mypy --help
-    - stubgen --help
-    - dmypy --help
-    - stubtest --help
-    - dir %SP_DIR%\mypy\*.pyd /s /b  # [win]
-    - ls $SP_DIR/mypy/*.so  # [linux and not (aarch64 and py==310)]
+outputs:
+  - name: mypy
+    build:
+      script:
+        - export MYPY_USE_MYPYC=1  # [not win]
+        - export SETUPTOOLS_USE_DISTUTILS=stdlib  # [not win]
+        - set MYPY_USE_MYPYC=1     # [win]
+        - set SETUPTOOLS_USE_DISTUTILS=stdlib  # [win]
+        # mypy searches for <>-stubs in site.getsitepackages()
+        # which points to $SP_DIR, even in cross-python
+        - cp -rv $BUILD_PREFIX/lib/python$PY_VER/site-packages/*-stubs $SP_DIR  # [build_platform != target_platform]
+        - cp -rv $BUILD_PREFIX/lib/python$PY_VER/site-packages/tomli* $SP_DIR  # [build_platform != target_platform]
+        - {{ PYTHON }} -m pip install . --no-deps -vv
+      entry_points:
+        - mypy = mypy.__main__:console_entry
+        - stubgen = mypy.stubgen:main
+        - dmypy = mypy.dmypy.client:console_entry
+        - stubtest = mypy.stubtest:main
+        # CLI provided here, but doesn't depend on a compiler
+        - mypyc = mypyc.__main__:main
+
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - {{ compiler('c') }}
+        # mypy's build imports this module, which has a C extension,
+        # so it has to be native to the build platform
+        - typed-ast >=1.4.0,<2                   # [build_platform != target_platform and py<38]
+        - tomli >=1.1.0
+      host:
+        - python
+        - pip
+        - mypy_extensions >=0.4.3
+        - typed-ast >=1.4.0,<2  # [py<38]
+        - typing_extensions >=3.10
+        - types-setuptools
+        - types-psutil
+        - types-typed-ast >=1.5.8,<1.6.0
+        - tomli >=1.1.0
+      run:
+        - python
+        - mypy_extensions >=0.4.3
+        - typed-ast >=1.4.0,<2  # [py<38]
+        - typing_extensions >=3.10
+        - tomli >=1.1.0
+        # extra: dmypy
+        - psutil >=4.0
+
+    test:
+      requires:
+        - pip
+      imports:
+        - mypy
+        - mypy.dmypy
+        - mypy.plugins
+        - mypy.server
+        - mypyc
+      commands:
+        - pip check
+        - dmypy --help
+        - mypy --help
+        - mypyc --help
+        - stubgen --help
+        - stubtest --help
+        - dir %SP_DIR%\mypy\*.pyd /s /b  # [win]
+        - ls $SP_DIR/mypy/*.so  # [linux and not (aarch64 and py==310)]
+
+  - name: mypyc
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage("mypy", exact=True) }}
+        - gcc_linux-64     # [linux]
```

Comment:
or maybe the suggestion of using the `compilers` package? how much does that bring along?

Reply:
ah, `cxx` and `fortran`... while very cool, `mypyc` wouldn't know what to do with those... would just `c-compiler` work?

Reply:
tyring `c-compiler` on https://github.com/conda-forge/mypyc-ipython-feedstock/pull/2

Reply:
looks good over there, will update here to use `c-compiler`

Reply:
Hey, I did not know we had a c-compiler package! Nice!

Reply:
Hmm:

does `mypyc` need a `{{ compiler('c') }}` in the `build` so it syncs up the versions correctly, or will the dep on `mypy` take care of that?

```
{
  "mypy-0.990-py38hdacc76a_2": {
    "recipe": {
      "build_platform": "linux-64",
      "c_compiler": "gcc",
      "c_compiler_version": "10",
      "channel_targets": "conda-forge main",
      "python": "3.8.* *_cpython",
      "target_platform": "linux-aarch64"
    }
  },
  "mypyc-0.990-py38ha081722_2": {
    "recipe": {
      "c_compiler": "gcc",
      "channel_targets": "conda-forge main",
      "python": "3.8.* *_cpython",
      "target_platform": "linux-aarch64"
    }
  }
}
```

Reply:
Great, all :heavy_check_mark: : other than this question about `build` deps to get the right version of the compiler, and there's not much else to do here.i

Reply:
> does mypyc need a {{ compiler('c') }} in the build so it syncs up the versions correctly, or will the dep on mypy take care of that?

In theory I think we are OK. If compilers is updated the c-compiled will be update too and things should sync up.

Reply:
Great. I'm willing to wait and see. At present `mypyc` is still kinda niche, and if you used it in a build, you'd still need to add `{{ compiler('c') }}` presumably, so it's really just the interactive case that would be "exciting". Think we're good here!